src/gei/Commands/MigrateRepoCommand.cs

```diff
@@ -454,6 +472,54 @@ private void LogAndValidateOptions(MigrateRepoCommandArgs args)
                 _log.LogInformation($"GIT ARCHIVE URL: {args.GitArchiveUrl}");
                 _log.LogInformation($"METADATA ARCHIVE URL: {args.MetadataArchiveUrl}");
             }
+            if (args.LfsMappingFile != "")
+            {
+                _log.LogInformation($"LFS MAPPING FILE: {args.LfsMappingFile}");
+            }
+        }
+
+        private byte[] applyLfsMappingFileToMetadata(byte[] metadataArchiveContent, string lfsMappingFile)
+        {
+            // extract the metadata archive
+            var gzipDataStream = new MemoryStream(metadataArchiveContent);
+            Stream inStream = new GZipInputStream(gzipDataStream);
+            TarArchive archive = TarArchive.CreateInputTarArchive(inStream, TarBuffer.DefaultBlockFactor);
+            archive.ExtractContents("./archiveExtracted");
+            
+            // modify the pull_requests_*.json files in the extracted archive
+            string[] fileNames = System.IO.Directory.GetFiles("./archiveExtracted");
+            foreach (string fileName in fileNames)
+            {
+                string text = File.ReadAllText(fileName);
+                var lfsMappingLines = File.ReadLines(lfsMappingFile);
+                foreach (var lfsMappingLine in lfsMappingLines)
+                {
+                    string[] lfsMapping = lfsMappingLine.Split(','); 
+                    text = text.Replace(lfsMapping[0], lfsMapping[1]);
+                }
+                File.WriteAllText(fileName, text);
+            }
+
+            // tar gz the modified archive
+            // TODO: figure out how to put this into a byte array rather than a file!!
+            Stream outStream = File.Create("newArchive.tar.gz");
+            outStream = new GZipOutputStream(outStream);
+            TarArchive newArchive = TarArchive.CreateOutputTarArchive(outStream, TarBuffer.DefaultBlockFactor);
+            newArchive.RootPath = "./archiveExtracted";
+            
+            if (fileNames.Length > 0) {
+                foreach (string name in fileNames) {
+                    TarEntry entry = TarEntry.CreateEntryFromFile(name);
+                    newArchive.WriteEntry(entry, true);
+                }
+            } else {
+                _log.LogInformation("No files found in archiveExtracted directory");
+            }
```

Comment:
This can be simplified:
```suggestion
            if (!fileNames.Any()) 
            {
                _log.LogInformation("No files found in archiveExtracted directory");
            }
            
            foreach (string name in fileNames) 
            {
                    TarEntry entry = TarEntry.CreateEntryFromFile(name);
                    newArchive.WriteEntry(entry, true);
            }
```